pkg/controller/managedresource/merger.go

```diff
@@ -91,6 +94,14 @@ func merge(origin string, desired, current *unstructured.Unstructured, forceOver
 		delete(newObject.Object, "status")
 	}
 
+	annotations := desired.GetAnnotations()
+	if annotations != nil && annotations[annotationKeyReplicas] == "true" {
+		preserveReplicas = true
+	}
+	if annotations != nil && annotations[annotationKeyResources] == "true" {
+		preserveResources = true
+	}
```

Comment:
You can omit the `nil` check, see https://play.golang.org/p/g4UUTCE7_e8

```suggestion
	if annotations[annotationKeyReplicas] == "true" {
		preserveReplicas = true
	}
	if annotations[annotationKeyResources] == "true" {
		preserveResources = true
	}
```